src/components/KTable/KTable.vue

```diff
@@ -1040,6 +1066,18 @@ const getTestIdString = (message: string): string => {
   return msg
 }
 
+watch([columnVisibility, tableHeaders], (newVals) => {
+  const newVisibility = newVals[0]
+  const newHeaders = newVals[1]
+  const newVisibleHeaders = newHeaders.filter((header: TableHeader) => newVisibility[header.label] !== false)
+
+  if (JSON.stringify(newVisibleHeaders) !== JSON.stringify(visibleHeaders.value)) {
+    emitTablePreferences()
+  }
+
+  visibleHeaders.value = newVisibleHeaders
+}, { deep: true, immediate: true })
```

Comment:
question: why does this need to be `immediate`? I wouldn't think it could change on mount?

Reply:
Because `visibleHeaders` needs to be initialized on mount once `tableHeaders` value is set.